sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -88,6 +91,11 @@ def test_doit_nonsquare():
     raises(ShapeError, lambda: MatPow(X, pi).doit())
 
 
+def test_doit_equals_pow(): #17179
+    X = ImmutableMatrix ([[1,0],[0,1]])
+    assert MatPow(X, n).doit() == X**n
```

Comment:
We should have `== X` at the end of this assert. Otherwise the test passes even if both don't evaluate